engine/src/witness/btc.rs

```diff
@@ -134,9 +132,20 @@ where
 		.logging("pre-witnessing")
 		.spawn(scope);
 
+	let btc_safety_margin = state_chain_client
+		.storage_value::<pallet_cf_ingress_egress::WitnessSafetyMargin<
+			state_chain_runtime::Runtime,
+			state_chain_runtime::BitcoinInstance,
+		>>(state_chain_stream.cache().hash)
+		.await?
+		// Default to berghain in case the value is missing (e.g. during initial upgrade)
+		.unwrap_or(BITCOIN_SAFETY_MARGIN);
```

Comment:
Hmm, this has the same issue of which network we're on I guess - perhaps we should do the same thing as we did in the migrations and use the NetworkEnvironment, and then default based on that, else we risk some issues when the update is down on perseverance, 2/3 block reorgs on bitcoin testnet are fairly common